File: _code-samples/issue-credentials/py/decode_hex.py
from binascii import unhexlify

def decode_hex(s_hex):
    """
    Try decoding a hex string as ASCII; return the decoded string on success,
    or the un-decoded string prefixed by '(BIN) ' on failure.
    """
    try:
        s = unhexlify(s_hex).decode("ascii")
        # Could use utf-8 instead, but it has more edge cases.
        # Optionally, sanitize the string for display before returning
    except:
        s = "(BIN) "+s_hex
    return s

File: _code-samples/issue-credentials/js/decode_hex.js
export function decodeHex(sHex) {
  /**
   * Try decoding a hex string as ASCII; return the decoded string on success,
   * or the un-decoded string prefixed by '(BIN) ' on failure.
   */
  try {
    const buffer = Buffer.from(sHex, "hex");
    return buffer.toString("ascii");
    // Could use utf-8 instead, but it has more edge cases.
    // Optionally, sanitize the string for display before returning
  } catch (err) {
    return "(BIN) " + sHex;
  }
}
